File: .anonymizer/ts-transformers/user-structure-fonction.py
#!/usr/bin/env python3
from datetime import datetime
import json
import random
import signal
import sys
from uuid import uuid4

signal.signal(signal.SIGTERM, lambda sig, frame: exit(0))

FONCTIONS = (
        "Agent administratif",
        "Agent d'accueil",
        "Educateur Spécialisé",
        "Président",
        "Responsable de service",
        "Travailleur social",
        None,
        None,
        None,
)

for json_line in sys.stdin:
    line = json.loads(json_line)
    new_fonction = FONCTIONS[random.randint(0, len(FONCTIONS) - 1)] 

    if new_fonction is None:
        line.update({"fonction": {"d": "", "n": True}})
    else:
        line.update({"fonction": {"d": new_fonction, "n": False}}) 

    sys.stdout.write(json.dumps(line) + "\n")
    sys.stdout.flush()

File: .anonymizer/ts-transformers/user-structure-fonction.mjs
#!/usr/bin/env -S node

import { main } from "./lib.mjs";
import { fromList } from "./data-helpers.mjs";

const stdout = process.stdout
const stderr = process.stderr

function anonymize(line) {

  const new_fonction = fromList([
    "Agent administratif",
    "Agent d'accueil",
    "Educateur Spécialisé",
    "Président",
    "Responsable de service",
    "Travailleur social",
    null,
    null,
    null,
  ])

  if (new_fonction === null) {
    line.fonction.d = ""
    line.fonction.n = true
  } else {
    line.fonction.d = new_fonction
    line.fonction.n = false
  }
  // stderr.write("Output: " + JSON.stringify(line) + "\n");
  stdout.write(JSON.stringify(line) + "\n");
}

main(anonymize)
